### PR TITLE
Add friendly error message for missing dotfiles

### LINKED ISSUE #544: TypeError: Cannot convert undefined or null to object in task lint
Download 1.2 directly, then did npm install and bower install, then do "gulp serve", and here is the error output:
```
[12:37:13] Starting 'lint'...
[12:37:14] 'lint' errored after 1.18 s
[12:37:14] TypeError: Cannot convert undefined or null to object
    at Function.keys (native)
    at copyConfiguration (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/config/configuration.js:1181:12)
    at NodeConfiguration.Configuration._processConfig (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/config/configuration.js:474:5)
    at NodeConfiguration.Configuration.load (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/config/configuration.js:238:10)
    at StringChecker.configure (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/string-checker.js:67:29)
    at Checker.configure (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/checker.js:28:39)
    at Object.module.exports (/Users/cnbuff410/node_modules/gulp-jscs/index.js:35:10)
    at Gulp.<anonymous> (/Users/cnbuff410/gulpfile.js:124:11)
    at module.exports (/Users/cnbuff410/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/cnbuff410/node_modules/gulp/node_modules/orchestrator/index.js:273:3)
```
It's caused by this line in lint task:

  .pipe($.jscs())

### PR DESCRIPTION
**Pull request by ohanhi**
Fixes #544.

This provides much more human-readable error messages for when certain dotfiles (namely `.jscsrc` or `.jshintrc`) are missing from the file system.

With the improvement:
screen shot 2015-12-02 at 23 26 38

Previously:
screen shot 2015-12-02 at 23 25 47

### CODE REVIEW

**Comment by ohanhi** [comment]
I signed it!

File: gulpfile.js [hunk 1]
```diff
@@ -105,8 +105,33 @@ gulp.task('elements', function() {
...
+// Check the files necessary for linting exist.
+// Without this, the 'lint' task can fail with a very non-descriptive message:
+// "TypeError: Cannot convert undefined or null to object"
+gulp.task('lintfilesExist', function(cb) {
+  var necessaryFiles = ['.jscsrc', '.jshintrc'];
+  var filesMissing = [];
+
+  var checked = 0;
+  necessaryFiles.forEach(function(file) {
+    fs.stat(__dirname + '/' + file, function(err, stats) {
```
Comment (samccone) [inline gulpfile.js]: path.join() here :)

**Comment by ohanhi** [comment]
@samccone Is this ok?

File: gulpfile.js [hunk 2]
```diff
@@ -105,8 +105,34 @@ gulp.task('elements', function() {
...
 
+// Check the files necessary for linting exist.
+// Without this, the 'lint' task can fail with a very non-descriptive message:
+// "TypeError: Cannot convert undefined or null to object"
+gulp.task('lintfilesExist', function(cb) {
+  var necessaryFiles = ['.jscsrc', '.jshintrc'];
+  var filesMissing = [];
+
+  var checked = 0;
+  necessaryFiles.forEach(function(file) {
```
Comment (samccone) [inline gulpfile.js]: can we do 

necessaryFiles.map(function(v){ return path.join(__dirname, v); })

reduce or filter down

then do the error check?


(I think in this case it is totally ok to use statSync just to keep the complexity low)

**Comment by samccone** [comment]
@ohanhi just ping me if you have any questions or whatever, or let me know any thoughts

thanks again for this great PR!

**Comment by robdodson** [comment]
I'd like to offer a dissenting opinion here. The gulpfile for this project is already extremely complicated and due for some house cleaning. I would vote against adding another task to check for missing files. We have a record of the problem and the fix documented in #544.

**Comment by samccone** [comment]
The flip side is that this makes an error, which several people now have hit, into a human readable easy to debug locally issue as compared to filing an issue on the bug tracker.

For that reason I am :+1: to it. Most people never touch the gulp file, and those that do are applying recipes.. which cause issues for other reasons .. but we have an idea how to fix that set of issues https://github.com/PolymerElements/polymer-starter-kit/issues/552

so, +1 to better errors --- the complexity argument is valid however I think the only reason why I see it as an issue are for the reasons state in 552

**Comment by robdodson** [comment]
I think this can be improved with a few lines of instructions in the README vs adding code to the project to support it.

File: gulpfile.js [hunk 3]
```diff
@@ -105,8 +105,34 @@ gulp.task('elements', function() {
   return styleTask('elements', ['**/*.css']);
 });
 
+// Check the files necessary for linting exist.
+// Without this, the 'lint' task can fail with a very non-descriptive message:
+// "TypeError: Cannot convert undefined or null to object"
+gulp.task('lintfilesExist', function(cb) {
```
Comment (robdodson) [inline gulpfile.js]: Instead of creating a separate task for this, could we move the code inside the lint task itself? @samccone wdyt?

Comment (robdodson) [inline gulpfile.js]: I did a variation on this which seems to work ok: https://gist.github.com/robdodson/9219e1b5042eeca14923

I'm not an expert in throwing errors in Gulp but here's what the output looks like:

screen shot 2015-12-06 at 9 27 18 am

Comment (samccone) [inline gulpfile.js]: personally I think a separate task is pretty nice, easier to test... debug ... ect ect

**Comment by samccone** [comment]
moved to #580

**Pull request closed without merging**